pallets/parachain-staking/src/lib.rs

```diff
@@ -1354,8 +1576,26 @@ pub mod pallet {
 					if x.amount > next {
 						Some(x)
 					} else {
-						if let Some(state) = <CollatorState<T>>::get(&x.owner) {
-							for bond in state.nominators.0 {
+						if let Some(state) = <CollatorState2<T>>::get(&x.owner) {
+							// return all top nominations
+							for bond in state.top_nominators {
+								// return stake to nominator
+								T::Currency::unreserve(&bond.owner, bond.amount);
+								// remove nomination from nominator state
+								if let Some(mut nominator) = <NominatorState<T>>::get(&bond.owner) {
```

Comment:
We're expecting that the nominator should always exist here right? Otherwise something is inconsistent about the state.

If I'm right about that then I suggest
```rust
let mut nominator = NominatorState::<T>::get(&bond.owner).expect("Collator state and nominator state are consistent. Collator state has a record of this nomination. Therefore, NominatorState also has a record. qed.");
```

Reply:
This is the case. The proposed change will mean the code will panic here if state is inconsistent. Before, it would've just skipped over it (so the error would've been silent).

I agree with the suggestion. Hopefully it never panics...